Sources/EventSource/EventParser.swift

```diff
@@ -9,12 +9,13 @@
 import Foundation
 
 public protocol EventParser: Sendable {
-    func parse(_ data: Data) -> [EVEvent]
+    func parse(_ data: Data) async -> [EVEvent]
 }
 
 /// ``ServerEventParser`` is used to parse text data into ``ServerEvent``.
-public struct ServerEventParser: EventParser {
-    let mode: EventSource.Mode
+actor ServerEventParser: EventParser {
```

Comment:
What is the purpose of making this an actor?

Reply:
`ServerEventParser` has a mutable property `buffer`, so its better to use reference type such as class, then to provide exclusive access to modifying buffer property and ensuring data-race safety I make `ServerEventParser` an actor, which ensures only one thread at a time could access this data. But I am still new to those concepts in Swift so if you have a better solution in mind please share 🙂